geode-management/src/main/java/org/apache/geode/management/api/ClusterManagementOperationResult.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more contributor license
+ * agreements. See the NOTICE file distributed with this work for additional information regarding
+ * copyright ownership. The ASF licenses this file to You under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the License. You may obtain a
+ * copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package org.apache.geode.management.api;
+
+import java.util.concurrent.CompletableFuture;
+
+import com.fasterxml.jackson.annotation.JsonIgnore;
+
+import org.apache.geode.annotations.Experimental;
+import org.apache.geode.management.internal.CompletableFutureProxy;
+
+@Experimental
+public class ClusterManagementOperationResult<V extends JsonSerializable>
+    extends ClusterManagementResult {
+  public ClusterManagementOperationResult() {}
+
+  public ClusterManagementOperationResult(boolean success, String message) {
+    super(success, message);
+  }
+
+  public ClusterManagementOperationResult(StatusCode statusCode, String message) {
+    super(statusCode, message);
+  }
+
+  @JsonIgnore
+  private CompletableFuture<V> operationResult;
+
+  @JsonIgnore
+  public CompletableFuture<V> getFutureResult() {
+    if (operationResult instanceof CompletableFutureProxy)
+      ((CompletableFutureProxy) operationResult).startPolling();
```

Comment:
Could the call of startPolling be moved to a place that knows it is a CompletableFutureProxy?

Reply:
getFutureResult on the locator returns a CompletableFuture, while getFutureResult on the client returns the CompletableFutureProxy.  The only way we know if the user cares about the result is if they call getFutureResult().  The only other option to is to unconditionally start polling...

Reply:
Do you see a problem with unconditionally calling startPolling? Do you see a use case in which you would start the async op but never want to check if it completed.

Reply:
it's an ugly optimization but....